worker.go

```diff
@@ -32,34 +32,28 @@ const maxWorkerConsecutiveFailures = 6
 
 var (
 	watcherIsEnabled bool
-	workersReadyWG   sync.WaitGroup
 	workerShutdownWG sync.WaitGroup
-	workersAreReady  atomic.Bool
-	workersAreDone   atomic.Bool
 	workersDone      chan interface{}
 	workers          = make(map[string]*worker)
 )
 
 func initWorkers(opt []workerOpt) error {
 	workersDone = make(chan interface{})
-	workersAreReady.Store(false)
-	workersAreDone.Store(false)
 
 	for _, o := range opt {
 		worker, err := newWorker(o)
 		worker.threads = make([]*phpThread, 0, o.num)
 		if err != nil {
 			return err
 		}
-		workersReadyWG.Add(worker.num)
 		for i := 0; i < worker.num; i++ {
 			go worker.startNewWorkerThread()
 		}
+		for i := 0; i < worker.num; i++ {
+			<-worker.ready
```

Comment:
What is the advantage of using a chan instead of a WaitGroup here? With the previous logic multiple worker scripts were able to start their threads in parallel, now they have to start sequentially.

Reply:
> What is the advantage of using a chan instead of a WaitGroup here?

There are significant issues with using a WaitGroup. A WaitGroup makes sense if everything happened exactly once, but workers can be restarted, shutdown, migrate between threads, and any number of 'things.' These all basically prevent a WaitGroup from making sense. Using a channel is much easier to simply wait for them all to become available.

> With the previous logic multiple worker scripts were able to start their threads in parallel, now they have to start sequentially.

They still start in parallel. There's a loop somewhere in any implementation. In this case, it is in the application instead of a semaphore. In fact, this is basically a semaphore, which Go doesn't have.

Reply:
> There are significant issues with using a WaitGroup. A WaitGroup makes sense if everything happened exactly once, but workers can be restarted, shutdown, migrate between threads, and any number of 'things.' These all basically prevent a WaitGroup from making sense. Using a channel is much easier to simply wait for them all to become available.

I agree with you, in #1137 I also removed this specific WaitGroup. I guess it just doesn't make sense in the current implementation, since workers do indeed only start once.

> They still start in parallel. There's a loop somewhere in any implementation. In this case, it is in the application instead of a semaphore. In fact, this is basically a semaphore, which Go doesn't expose to developers.

They are starting in parallel for 1 worker script. What I meant is, they don't start in parallel for multiple worker scripts anymore.